docs/docs/genai/concepts/scorers.mdx

```diff
@@ -0,0 +1,230 @@
+    description: "Track quality metrics over time in production, detect degradations early, and maintain high standards as your application evolves and scales."
+  }
+]} />
+
+## Types of Scorers
+
+MLflow provides several types of scorers to address different evaluation needs:
+
+<ConceptOverview concepts={[
+  {
+    icon: Brain,
+    title: "LLM-based Scorers (LLM-as-a-Judge)",
+    description: "Use large language models to evaluate subjective qualities like helpfulness, coherence, and style. These scorers can understand context and nuance that rule-based systems miss."
+  },
+  {
+    icon: Code,
+    title: "Code-based Scorers",
+    description: "Custom Python functions for deterministic evaluation. Perfect for metrics that can be calculated algorithmically like ROUGE scores, exact match, or custom business logic."
+  },
+  {
+    icon: Activity,
+    title: "Agentic Judges",
+    description: "Autonomous agents that analyze execution traces to evaluate not just outputs, but the entire process. They can assess tool usage, reasoning chains, and error handling."
+  },
+  {
+    icon: Users,
+    title: "Human-Aligned Judges",
+    description: "LLM judges that have been fine-tuned with human feedback to match your specific quality standards. These provide the consistency of automation with the nuance of human judgment."
+  }
+]} />
+
+## Scorer Output Structure
+
+All scorers in MLflow produce standardized output that integrates seamlessly with the evaluation framework:
+
+| Field       | Type             | Description                                                                |
+| ----------- | ---------------- | -------------------------------------------------------------------------- |
+| `name`      | `str`            | Unique identifier for the scorer (e.g., "correctness", "safety")           |
+| `value`     | `Any`            | The evaluation result - can be numeric, boolean, or categorical            |
+| `rationale` | `Optional[str]`  | Explanation of why this score was given (especially useful for LLM judges) |
+| `metadata`  | `Optional[dict]` | Additional information about the evaluation (confidence, sub-scores, etc.) |
+| `error`     | `Optional[str]`  | Error message if the scorer failed to evaluate                             |
+
+## Common Scorer Patterns
+
+### Binary Classification
+
+```python
+# Simple pass/fail evaluation
+def is_safe(outputs):
+    # Check for harmful content
+    return "harmful" not in outputs.lower()
+```
+
+### Numeric Scoring
+
+```python
+# Score on a scale (e.g., 0-1 or 1-5)
+def relevance_score(inputs, outputs):
+    # Calculate relevance score
+    return calculate_similarity(inputs, outputs)
+```
+
+### Categorical Rating
+
+```python
+# Multi-class evaluation
+def quality_rating(outputs):
+    # Classify into categories
+    if len(outputs) < 50:
+        return "poor"
+    elif len(outputs) < 200:
+        return "good"
+    else:
+        return "excellent"
+```
```

Comment:
@BenWilson2 These feel a bit contrived. Can we do something like language perplexity / reading level?